### PR TITLE
Enhance Hystrix ThreadPool mode so that it can correctly pass Istio's call chain headers

### PR DESCRIPTION
**Pull request by idefav**
Enhance Hystrix ThreadPool mode so that it can correctly pass Istio's call chain headers, and set the RestTemplate interceptor to intercept ordinary Http requests, providing the ThreadPoolHolder operation class to manually synchronize custom thread pool variables.

Use the effect to string the entire call chain.
image
image

### CODE REVIEW

**Comment by geoand** [comment]
Very interesting! @salaboy would you like to take a look?

**Comment by spencergibb** [comment]
I think we're duplicating functionality from sleuth, no?

**Comment by idefav** [comment]
> I think we're duplicating functionality from sleuth, no?

This PR is not only enhanced for Hystrix, but also for RestTemplate, Ribbon, Feign, custom thread pool and HTTPClient. The Hystrix thread pool mode does not support the call chain. Here you need to enhance it to support Istio's Headers. pass.

**Comment by spencergibb** [comment]
Sleuth instruments all those things

File: spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/trace/FeignPropagationRequestInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+@ConditionalOnClass(RequestInterceptor.class)
+@ConditionalOnProperty(value = "spring.cloud.istio.enabled", matchIfMissing = true)
+public class FeignPropagationRequestInterceptor implements RequestInterceptor {
+
+	private static final Logger LOG = LoggerFactory
+			.getLogger(FeignPropagationRequestInterceptor.class);
+
+	private static List<String> HEADERS = Arrays.asList(IstioConstants.X_REQUEST_ID,
+			IstioConstants.X_B3_TRACEID, IstioConstants.X_B3_SPANID,
```
Comment (adriancole) [inline spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/trace/FeignPropagationRequestInterceptor.java]: similarly missing "b3"

File: spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/trace/IstioHeaderPropagatoinRequestFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * be passed.
+ *
+ * @author wuzishu
+ * @see org.springframework.cloud.kubernetes.istio.trace.HeaderPropagationRequestFilter
+ */
+@ConditionalOnProperty(value = "spring.cloud.istio.enabled", matchIfMissing = true)
+public class IstioHeaderPropagatoinRequestFilter extends HeaderPropagationRequestFilter {
+
+	private static List<String> HEADERS = Arrays.asList(IstioConstants.X_REQUEST_ID,
+			IstioConstants.X_B3_TRACEID, IstioConstants.X_B3_SPANID,
```
Comment (adriancole) [inline spring-cloud-kubernetes-istio/src/main/java/org/springframework/cloud/kubernetes/istio/trace/IstioHeaderPropagatoinRequestFilter.java]: "b3"

**Comment by spencergibb** [comment]
I'm going to close this. If sleuth is missing anything, it should be added there.

**Pull request closed without merging**